entity/src/advisory.rs

```diff
@@ -9,6 +10,10 @@ pub struct Model {
     pub location: String,
     pub sha256: String,
     pub title: Option<String>,
+    pub aggregate_severity: Option<String>,
+
+    /// Holds the date when the current revision of this document was released.
+    pub current_release_date: Option<DateTime<Utc>>,
```

Comment:
IIRC current release date is a mandatory field, we should map it accordingly.

Reply:
Thanks for pointing this out. It allowed me to make 2 observations:
- @bobmcwhirter you wrote this block of code where we are ingesting a CVE file into the `Advisory` model (db table) https://github.com/trustification/trustify/blob/main/ingestors/src/cve/loader.rs#L54-L56 . There is also a test that points to this file https://github.com/trustification/trustify/blob/main/etc/test-data/mitre/CVE-2024-28111.json
  - My question is what exactly are we going to ingest under the `advisory` abstraction (model/table)? If the current `main` branch is correct then: @ctron I'm afraid that although `current_release_date` is a mandatory field for a CSAF file, it does not even exist on CVE files.

```rust
        self.graph
            .ingest_advisory(cve.cve_metadata.cve_id(), location, sha256, &tx)
            .await?;
```

- The same applies to OSV files https://github.com/trustification/trustify/blob/main/ingestors/src/advisory/osv/loader.rs#L48-L50 . We are ingesting them as an advisory.
  - This is the file ingested as an advisory https://github.com/trustification/trustify/blob/main/etc/test-data/osv/RUSTSEC-2021-0079.json 

I know `trustify` won't necessarily be exactly the same as `Trustification` but in Trustification we strictly stored `CSAF` files under the `Advisory` model. Am I right @ctron ?

May I ask for clarification on this regard please. The image below is what an advisory mean for Trustification. And advisory has `aggregated severity` + `Title` + `a set of Vulnerabilities associated to it` + `information about products affected by that advisory`

![Screenshot from 2024-03-17 12-28-55](https://github.com/trustification/trustify/assets/2582866/7c691fa4-7faa-41d5-953b-0d3c66e67cd9)

Reply:
Alright. It seems, Advisories will store CSAF + CVEs. Therefore `metadata` information should be defines as something that all of them have and not specific to a single schema. So far, in this PR I've defined 3 things that CSAF + CVES or any other file should have:
```rust
    pub title: Option<String>,
    pub severity: Option<String>,
    /// Holds the date when the current revision of this document was released.
    pub release_date: Option<DateTime<Utc>>,
```

- I renamed `aggregated_severity` to `severity` to not be CSAF specific.
- I renamed `current_release_data` to `release_date`.
- I'm forcing the metadata to be set when an advisory is instantiated (inserted into db).
- This PR is implementing the code for setting the metadata for CSAF files, but I'll open another PR to populate the same metadata for CVEs

Let me know what you think about it.

Reply:
This is looking nicer for sure.

fwiw, a single Advisory can contain information about multiple Vulnerabilities.

Currently any score is on the Advisory, but perhaps should be on the join of Advisory+Vulnerabililty, to accomodate more of the CSAF model.